lib/ts/recipe/emailverification/types.ts

```diff
@@ -23,8 +23,9 @@ import EmailDeliveryIngredient from "../../ingredients/emaildelivery";
 import { GeneralErrorResponse } from "../../types";
 
 export type TypeInput = {
+    mode: "REQUIRED" | "OPTIONAL" | "OFF";
     emailDelivery?: EmailDeliveryTypeInput<TypeEmailVerificationEmailDeliveryInput>;
-    getEmailForUserId: (userId: string, userContext: any) => Promise<string>;
+    getEmailForUserId?: (userId: string, userContext: any) => Promise<string>;
     getEmailVerificationURL?: (user: User, userContext: any) => Promise<string>;
```

Comment:
While we are at it, maybe we don't need this config anymore - cause you can change the URL or any part of the URL fairly easily by overriding the emailDelivery config.